fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -1157,6 +1187,167 @@ long getTotalRecordsScanned() {
+        @API(API.Status.UNSTABLE)
+        public static class Builder {
+            private int maxLimit = DEFAULT_LIMIT;
+            private int maxRetries = DEFAULT_MAX_RETRIES;
+            private int recordsPerSecond = DEFAULT_RECORDS_PER_SECOND;
+            private long progressLogIntervalMillis = DEFAULT_PROGRESS_LOG_INTERVAL;
+            private int increaseLimitAfter = DO_NOT_RE_INCREASE_LIMIT;
+
+            protected Builder() {
+
+            }
+
+            /**
+             * Set the maximum number of records to process in one transaction.
+             *
+             * The default limit is {@link #DEFAULT_LIMIT} = {@value #DEFAULT_LIMIT}.
+             * @param limit the maximum number of records to process in one transaction
+             * @return this builder
+             */
+            @Nonnull
+            public Builder setLimit(int limit) {
+                this.maxLimit = limit;
+                return this;
+            }
+
+            /**
+             * Set the maximum number of times to retry a single range rebuild.
+             *
+             * The default number of retries is {@link #DEFAULT_MAX_RETRIES} = {@value #DEFAULT_MAX_RETRIES}.
+             * @param maxRetries the maximum number of times to retry a single range rebuild
+             * @return this builder
+             */
+            @Nonnull
+            public Builder setMaxRetries(int maxRetries) {
+                this.maxRetries = maxRetries;
+                return this;
+            }
+
+            /**
+             * Set the maximum number of records to process in a single second.
+             *
+             * The default number of retries is {@link #DEFAULT_RECORDS_PER_SECOND} = {@value #DEFAULT_RECORDS_PER_SECOND}.
+             * @param recordsPerSecond the maximum number of records to process in a single second.
+             * @return this builder
+             */
+            @Nonnull
+            public Builder setRecordsPerSecond(int recordsPerSecond) {
+                this.recordsPerSecond = recordsPerSecond;
+                return this;
+            }
+
+
+            /**
+             * Set the minimum time between successful progress logs when building across transactions.
+             * Negative will not log at all, 0 will log after every commit.
+             *
+             * @param progressLogIntervalMillis the number of milliseconds to wait between successful logs
+             * @return this builder
+             */
+            @Nonnull
+            public Builder setProgressLogIntervalMillis(long progressLogIntervalMillis) {
+                this.progressLogIntervalMillis = progressLogIntervalMillis;
+                return this;
+            }
+
+            /**
+             * Set the number of successful range builds before re-increasing the number of records to process in a single
+             * transaction. The number of records to process in a single transaction will never go above {@link #limit}.
```

Comment:
Should that be `maxLimit` instead of `limit`? I'm also not sure that link would work if it's referencing a private variable.

Reply:
done.